javascript/javascript3/week3/shopping-cart.js

```diff
@@ -0,0 +1,95 @@
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+  //currencyconverter
+  convertToCurrency(currency) {
+    fetch(
+      `https://v6.exchangerate-api.com/v6/5b946308252733231c2f83fe/latest/${currency}`
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        const conversionRate = data.conversion_rates.DKK;
+        return this.price / conversionRate;
+      });
+  }
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+    // Implement functionality here
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    const index = this.products.indexOf(product);
+    if (index > -1) {
+      // only splice array when item is found
+      this.products.splice(index, 1);
+    }
+  }
+
+  searchProduct(productName) {
+    // Implement functionality here
+    const result = this.products.filter((product) => {
+      return product.name === productName;
+    });
+  }
+
+  getTotal() {
+    // Implement functionality here
+    const sum = this.products.reduce((accumulator, object) => {
+      return accumulator + object.price;
+    }, 0);
+    const renderedProducts = document.getElementById("products");
+    const p = document.createElement("p");
+    p.innerHTML = `Total sum: ${sum}`;
+    renderedProducts.appendChild(p);
+  }
+
```

Comment:
Here we can simply  return total value as ' return  this.products.reduce((accumulator, object) => { accumulator + object.price}, 0)'
and DOM functionality in the renderProducts function with this.getTotal()